sa/storage-authority.go

```diff
@@ -242,13 +252,19 @@ func (ssa *SQLStorageAuthority) NewRegistration(reg core.Registration) (core.Reg
 	if err != nil {
 		return reg, err
 	}
-
-	err = tx.Insert(&reg)
+	rm, err := registrationToModel(&reg)
+	if err != nil {
+		return reg, err
+	}
+	err = tx.Insert(rm)
 	if err != nil {
 		tx.Rollback()
 		return reg, err
 	}
-
+	reg, err = modelToRegistration(rm)
+	if err != nil {
+		return reg, err
```

Comment:
Looking at the code, this appears to be a holdover from a time when we generated registration ids as random tokens, and had a loop to check if any token already existed. It was probably unnecessary even then: A unique index would have prevented the insert in the (vanishingly unlikely) event there was a duplicate. Now we manage ids with autoincrement anyhow. So I'd vote to just remove the transaction code from this method.